test/integration/accounting.ts

```diff
@@ -0,0 +1,812 @@
+    expect(postTotalSharesEvent[0].args.preTotalPooledEther).to.equal(
+      postTotalSharesEvent[0].args.postTotalPooledEther + amountOfETHLocked,
+    );
+
+    const ethBalanceAfter = await ethers.provider.getBalance(lido.address);
+    expect(ethBalanceBefore).to.equal(ethBalanceAfter + amountOfETHLocked);
+  });
+
+  it("Should account correctly with negative CL rebase", async () => {
+    const { lido, accountingOracle } = ctx.contracts;
+
+    const REBASE_AMOUNT = ether("-1000");
+
+    const lastProcessingRefSlotBefore = await accountingOracle.getLastProcessingRefSlot();
+    const totalELRewardsCollectedBefore = await lido.getTotalELRewardsCollected();
+    const totalPooledEtherBefore = await lido.getTotalPooledEther();
+    const totalSharesBefore = await lido.getTotalShares();
+
+    // Report
+    const { reportTx } = await report(ctx, { clDiff: REBASE_AMOUNT, excludeVaultsBalances: true });
+    const reportTxReceipt = (await reportTx!.wait()) as ContractTransactionReceipt;
+
+    const { amountOfETHLocked, sharesBurntAmount } = getWithdrawalParams(reportTxReceipt);
+
+    const lastProcessingRefSlotAfter = await accountingOracle.getLastProcessingRefSlot();
+    expect(lastProcessingRefSlotBefore).to.be.lessThan(lastProcessingRefSlotAfter);
+
+    const totalELRewardsCollectedAfter = await lido.getTotalELRewardsCollected();
+    expect(totalELRewardsCollectedBefore).to.equal(totalELRewardsCollectedAfter);
+
+    const totalPooledEtherAfter = await lido.getTotalPooledEther();
+    expect(totalPooledEtherBefore + REBASE_AMOUNT).to.equal(totalPooledEtherAfter + amountOfETHLocked);
+
+    const totalSharesAfter = await lido.getTotalShares();
+    expect(totalSharesBefore).to.equal(totalSharesAfter + sharesBurntAmount);
+
+    const tokenRebasedEvent = getEvents(reportTxReceipt, "TokenRebased");
+    const { sharesRateBefore, sharesRateAfter } = shareRateFromEvent(tokenRebasedEvent[0]);
+    expect(sharesRateAfter).to.be.lessThan(sharesRateBefore);
+
+    const ethDistributedEvent = getEvents(reportTxReceipt, "ETHDistributed");
+    expect(ethDistributedEvent[0].args.preCLBalance + REBASE_AMOUNT).to.equal(
+      ethDistributedEvent[0].args.postCLBalance,
+    );
+
+    const postTotalSharesEvent = getEvents(reportTxReceipt, "PostTotalShares");
+    expect(postTotalSharesEvent[0].args.preTotalPooledEther + REBASE_AMOUNT).to.equal(
+      postTotalSharesEvent[0].args.postTotalPooledEther + amountOfETHLocked,
+    );
+  });
+
+  it("Should account correctly with positive CL rebase close to the limits", async () => {
+    const { lido, accountingOracle, oracleReportSanityChecker, stakingRouter } = ctx.contracts;
+
+    const annualIncreaseLimit = (await oracleReportSanityChecker.getOracleReportLimits())[2];
+    const preCLBalance = (await lido.getBeaconStat()).slice(-1)[0];
+
+    let rebaseAmount = (preCLBalance * annualIncreaseLimit * (ONE_DAY + 1n)) / (365n * ONE_DAY) / MAX_BASIS_POINTS;
+    rebaseAmount = roundToGwei(rebaseAmount);
```

Comment:
I believe it's important to include a detailed description of our rebase amount calculation. Future colleagues (or yourself) will appreciate the clarity.

For example.

```suggestion
    // To calculate the rebase amount close to the annual increase limit
    // we use (ONE_DAY + 1n) to slightly underperform for the daily limit
    // This ensures we're testing a scenario very close to, but not exceeding, the annual limit
    let rebaseAmount = (preCLBalance * annualIncreaseLimit * (ONE_DAY + 1n)) / (365n * ONE_DAY) / MAX_BASIS_POINTS;
    rebaseAmount = roundToGwei(rebaseAmount);

    // At this point, rebaseAmount represents a positive CL rebase that is
    // just slightly below the maximum allowed daily increase, testing the system's
    // behavior near its operational limits
```